### PR TITLE
Refactor trashcan open event

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #4203

### Proposed Changes

Creates dedicated class (`Blockly.Events.TrashcanOpen`) for trashcan open events.

### Reason for Changes

See issue, but generally motivated by:
- better typing
- clearer events

### Test Coverage

Tested by logging events on playground and observing that expected events were logged and running mocha tests.

Tested on:
 * Desktop Chrome

### CODE REVIEW

File: core/ui_events.js [hunk 1]
```diff
@@ -113,5 +114,51 @@ Blockly.Events.Click.prototype.fromJson = function(json) {
   this.blockId = json['blockId'];
 };
 
+/**
+ * Class for a trashcan open event.
+ * @param {boolean=} opt_isOpen Whether the trashcan flyout is opening.
```
Comment (alschmiedt) [inline core/ui_events.js:119]: Should this be 'is open'?

File: core/ui_events.js [hunk 2]
```diff
@@ -113,5 +114,51 @@ Blockly.Events.Click.prototype.fromJson = function(json) {
...
+   */
+  this.isOpen = opt_isOpen;
+};
+Blockly.utils.object.inherits(Blockly.Events.TrashcanOpen, Blockly.Events.Ui);
+
+/**
+ * Type of this event.
+ * @type {string}
+ */
+Blockly.Events.TrashcanOpen.prototype.type = Blockly.Events.TRASHCAN_OPEN;
```
Comment (alschmiedt) [inline core/ui_events.js:140]: Does this constant get added in a different PR?

**Review by alschmiedt — APPROVED** [review]

File: core/ui_events.js (hunk 2 above)
Comment (moniika) [reply core/ui_events.js:140]: Yes

File: core/ui_events.js (hunk 1 above)
Comment (moniika) [reply core/ui_events.js]: Updated. I think 'is open' is easier to understand.

**Pull request merged**